src/core/execute.c

```diff
@@ -4667,7 +4704,7 @@ static int exec_child(
                 }
         }
 
-        if (needs_sandboxing && context->private_users && !have_effective_cap(CAP_SYS_ADMIN)) {
+        if (needs_sandboxing && exec_context_need_unprivileged_private_users(context)) {
```

Comment:
"unprivileged"? why that? if this is applied to system services, there is no "unprivileged" involved?